packages/types/src/interfaces.ts

```diff
@@ -43,4 +43,205 @@ export interface SubstrateEvent extends EventRecord {
   block: SubstrateBlock;
 }
 
-export type DynamicDatasourceCreator = (name: string, args: Record<string, unknown>) => Promise<void>;
+export type DynamicDataSourceCreator = (name: string, args: Record<string, unknown>) => Promise<void>;
+
+export enum ETransactionType {
```

Comment:
Where possible we should not reimplement types provided by the algorand sdk.

e.g https://github.com/algorand/js-algorand-sdk/blob/57c5645676ed5bc8b91e3c6c19511c02b0a6337e/src/abi/transaction.ts#L3